official/resnet/resnet_run_loop.py

```diff
@@ -287,34 +309,37 @@ def exclude_batch_norm(name):
       eval_metric_ops=metrics)
 
 
-def validate_batch_size_for_multi_gpu(batch_size):
+def compute_per_device_batch_size(batch_size, use_distribution_strategy,
+                                  gpus_for_distribution_strategy):
   """For multi-gpu, batch-size must be a multiple of the number of GPUs.
 
-  Note that this should eventually be handled by replicate_model_fn
+  Note that this should eventually be handled by DistributionStrategies
   directly. Multi-GPU support is currently experimental, however,
   so doing the work here until that feature is in place.
 
   Args:
-    batch_size: the number of examples processed in each training batch.
+    batch_size: Global batch size to be divided among devices.
+    use_distribution_strategy: Whether DistributionStrategies API is used.
+    gpus_for_distribution_strategy: How many GPUs are used with
+      DistributionStrategies.
+
+  Returns:
+    Batch size per device.
 
   Raises:
-    ValueError: if no GPUs are found, or selected batch_size is invalid.
+    ValueError: if batch_size is not divisible by number of devices
   """
-  from tensorflow.python.client import device_lib  # pylint: disable=g-import-not-at-top
-
-  local_device_protos = device_lib.list_local_devices()
-  num_gpus = sum([1 for d in local_device_protos if d.device_type == 'GPU'])
```

Comment:
i wonder if this check is still useful in some form... for e.g. should we check that actual number of gpus available >= gpus_for_distribution_strategy

Reply:
I feel like that check should live under the hood in DistStrat-- we had to break our own rules and import from the private API here. Is it feasible to do that on the DistStrat side? Or just ignore for now and let the error bubble up when it is hit...